sheets/batch_get.go

```diff
@@ -117,6 +119,8 @@ func (g *BatchReader) valueRangesToRecords(valueRanges []*sheets.MatchedValueRan
 	for i := range valueRanges {
 		valueRange := valueRanges[i].ValueRange
 		values := valueRange.Values
+		// haris: what's `i` and what's `index` here?
+		// Maybe it's the same answer as for what's difference between valueRanges and valueRange.values?
```

Comment:
the 'i' and the 'index' both represents the array index, as variables with same names under nested loop would give unexpected results.
Done.

Reply:
Yeah, I got that part related to not using the same var name. I was asking more about what those indexes mean. 

`values := valueRanges[i].ValueRange.Values` appear to be the rows, and then `index, val` are actually the row number and the row itself?

Reply:
@hariso We are using `BatchGetByDataFilter` Sheets API to get the rows data using the sheet's gid instead of the sheet name. The API supports fetching data for multiple value ranges (like Sheet1!1:100 & Sheet1!200:300 & Sheet2!100-200), The ValueRanges array contains data for each of the requested Ranges.
`ValueRange.Values` correspond to the actual range data (i.e rows)

Reply:
Added link to the response body in comments https://github.com/gopherslab/conduit-connector-google-sheets/pull/3/files#diff-4c3f1e257f1d20c3e64bfd6d22aba81b1d7ba66588966716e396cb3e417e29a8R133-R135